### PR TITLE
add Save control

### PR DESCRIPTION
**Pull request by olivierlabayle**
This PR needs:

- [x] merge of https://github.com/JuliaAI/MLJBase.jl/pull/733 and release of MLJBase 0.20

This works locally with https://github.com/JuliaAI/MLJBase.jl/tree/serialization installed.

Todo:

- [x] Update MLJBase dep to 0.20
- [x]  Bump version to 0.5

### CODE REVIEW

File: src/MLJIteration.jl [hunk 1]
```diff
@@ -39,6 +40,7 @@ include("constructors.jl")
 include("traits.jl")
 include("ic_model.jl")
 include("core.jl")
+include("serialization.jl")
```
Comment (ablaom) [inline src/MLJIteration.jl:43]: This `include` logically needs to precede line 36, or else just dump the code in "controls.jl", which already contains the other controls added by this package. I'd prefer the latter, to be honest.

File: src/serialization.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
+struct Save
+    filename::String
+    method
+end
+
+Save(filename="machine.jlso"; method=serialize) =
+    Save(filename, method)
+
```
Comment (ablaom) [inline src/serialization.jl:8]: This doesn't quite look right. I want to be able to do `Save(filename="machine.jlso")` but that doesn't work. 

For consistency with existing controls, I think all the following should work:

`Save("machine.jlso")`
`Save("machine.jslo", method=...)`
`Save(filename="machine.jlso")`
`Save(filename="machine.jlso", method=...)`

File: src/serialization.jl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+IterationControl.@create_docs(Save,
+             header="Save(filename=\"machine.jlso\")",
+             example="Save(\"run3/machine.jlso\", compression=:gzip)",
+             body="Save the current state of the machine being iterated to "*
```
Comment (ablaom) [inline src/serialization.jl:13]: I see the `header` and `example` are not consistent - one uses `filename` as kwarg, other as regular arg. 

Okay, I guess you've not looked at this doc-strong yet.

File: test/serialization.jl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+using Serialization
+using MLJBase
+using MLJIteration
+using IterationControl
+using .DummyModel
+
+
+@testset "Save" begin
+    X, y = make_dummy(N=8);
+    @test_throws Exception Save(filename="myfile.jlso")
```
Comment (ablaom) [inline test/serialization.jl:15]: I guess it's confusing to still have .jlso extension here as this is now .jls.

**Review by ablaom — COMMENTED** [review]
I guess this is WIP. 

We probably want a test for `method=(filename, mach)->JSLO.save(filename, mach)` or some other format.

**Comment by olivierlabayle** [comment]
> I guess this is WIP.

Sorry, my bad I did not review my changes correctly before making the PR. I have taken your comments into consideration:

- Removed the `serialization.jl` file
- Updated docs
- Added a full kw constructor, and parametrized `Save` as for other controls.
- Added a JLSO test.

**Review by ablaom — APPROVED** [review]
Excellent, thanks.

**Pull request merged**